### PR TITLE
Disable exclusive triggers when there aren't any other types in play

### PR DESCRIPTION
**Pull request by msullivan**

### CODE REVIEW

File: tests/test_edgeql_update.py [hunk 1]
```diff
@@ -3419,6 +3419,16 @@ async def test_edgeql_update_cardinality_assertion(self):
                 }
             ''')
 
+    @test.xfail('''
```
Comment (msullivan) [inline tests/test_edgeql_update.py:3422]: We need to decide if this behavior change is unacceptable, or find some way around it.

Comment (elprans) [reply tests/test_edgeql_update.py:3422]: Well, we haven't really defined the behavior either way, so I think it's fine if the behavior changes like so.

**Review by elprans — APPROVED** [review]

File: tests/test_edgeql_update.py (hunk 1 above)
Comment (msullivan) [reply tests/test_edgeql_update.py:3422]: The catch is that the behavior then depends on whether or not there is a subtype, which is kind of hinky. Not necessarily disqualifying, though, since it's such an edge case. (And you could always opt in to the trigger-based behavior by creating a dummy subtype, if you somehow cared.)

Comment (elprans) [reply tests/test_edgeql_update.py:3422]: How hard would it be to repurpose the `unless conflict` machinery to handle constraints also and get rid of triggers altogether?

Comment (msullivan) [reply tests/test_edgeql_update.py:3422]: Currently, when used to raise errors, the `conflict` machinery makes sure to exclude all the data from the base relations from the queries it is doing. Before I did this, I think it was causing a 25% slowdown in the test suite, or some such.

That said, that perf trouble might be fixable, though it's a question mark. It would probably take a couple days to get working?

Would the idea be to try to always implement the "no trigger" style behavior?

Comment (elprans) [reply tests/test_edgeql_update.py:3422]: Getting rid of triggers would be ideal, yeah.

Comment (msullivan) [reply tests/test_edgeql_update.py:3422]: That's not what I mean with the question about the desired behavior, though. There are two possible semantics here:
 * Whether to raise an error is based on comparing to the state at the *start* of the statement. That is, deleting something and then adding something with the same key *still errors*. That's the semantics you get when using triggers.
 * The reverse, where if you delete something and then add something with the same key, it *works*. That's the semantics you get when triggers are off but there is a constraint on the table (it seems).

We would need to make a choice for which to implement

Comment (elprans) [reply tests/test_edgeql_update.py:3422]: Well, technically there isn't any causality between the statements, because the order of execution is undefined (unless you explicitly correlate the statements), so Option 1 seems to make more sense.

**Comment by msullivan** [comment]
Cherry-picked to 1.2

File: tests/test_edgeql_update.py (hunk 1 above)
Comment (msullivan) [reply tests/test_edgeql_update.py:3422]: We talked out of band and decided that for now, the behavior here is unspecified, and we can revisit it later. It's pretty marginal.
